pkg/codegen/hcl2/binder_schema.go

```diff
@@ -80,13 +80,13 @@ func (b *binder) loadReferencedPackageSchemas(n Node) error {
 	contract.Assert(len(diags) == 0)
 
 	for _, name := range packageNames.SortedValues() {
-		if _, ok := b.options.packageCache.entries[name]; ok {
+		if _, ok := b.referencedPackages[name]; ok {
```

Comment:
Huh, good catch. The broken code must have come from a bad refactoring or something--`referencedPackages` is supposed to have a peer set that performs this function. The map works, but can we sort the list returned by `Packages()` by name?

Reply:
I don't _think_ the old version was sorting them either: packages coming from different resources would be appended at resource order. But I added the sorting in `Packages()`. (OMG, is it really that hard in Go?)

Reply:
It wasn't sorting them per se, but the old order was deterministic. This looks good.